src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -565,77 +565,96 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
             }
         }
 
-        private async ValueTask<(Stream, TransportContext?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
+        private async ValueTask<(Stream, TransportContext?, Activity?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
         {
             Stream? stream = null;
             IPEndPoint? remoteEndPoint = null;
-            switch (_kind)
-            {
-                case HttpConnectionKind.Http:
-                case HttpConnectionKind.Https:
-                case HttpConnectionKind.ProxyConnect:
-                    stream = await ConnectToTcpHostAsync(_originAuthority.IdnHost, _originAuthority.Port, request, async, cancellationToken).ConfigureAwait(false);
-                    // remoteEndPoint is returned for diagnostic purposes.
-                    remoteEndPoint = GetRemoteEndPoint(stream);
-                    if (_kind == HttpConnectionKind.ProxyConnect && _sslOptionsProxy != null)
-                    {
-                        stream = await ConnectHelper.EstablishSslConnectionAsync(_sslOptionsProxy, request, async, stream, cancellationToken).ConfigureAwait(false);
-                    }
-                    break;
+            Exception? exception = null;
+            TransportContext? transportContext = null;
 
-                case HttpConnectionKind.Proxy:
-                    stream = await ConnectToTcpHostAsync(_proxyUri!.IdnHost, _proxyUri.Port, request, async, cancellationToken).ConfigureAwait(false);
-                    // remoteEndPoint is returned for diagnostic purposes.
-                    remoteEndPoint = GetRemoteEndPoint(stream);
-                    if (_sslOptionsProxy != null)
-                    {
-                        stream = await ConnectHelper.EstablishSslConnectionAsync(_sslOptionsProxy, request, async, stream, cancellationToken).ConfigureAwait(false);
-                    }
-                    break;
+            Activity? activity = ConnectionSetupDiagnostics.StartConnectionSetupActivity(IsSecure, OriginAuthority);
```

Comment:
Good catch, `ws/wss` may report `http/https` wrongly in both Metrics and Distributed Tracing. *If* we want to fix this, I would prefer to do it by opting out from creating an activity for those for now, and cover other protocols later.

IMHO in practice this will unlikely hurt anyone for an experimental feature, so my preference would be to create a tracking issue for .NET 10, where we could fix it for both Metrics and Tracing.

Reply:
>  it as-is for now, and create a tracking issue for .NET 10

:+1: